src/main/java/eu/dissco/core/digitalspecimenprocessor/service/ProcessingService.java

```diff
@@ -1,632 +1,417 @@
-              .toList());
-    } catch (DataAccessException e) {
-      log.error("Unable to update records into database. Rolling back updates", e);
-      rollbackService.rollbackUpdatedSpecimens(digitalSpecimenRecords, mediaPidMap, false, false);
-      return Collections.emptySet();
+    if (!specimenProcessResult.newSpecimens().isEmpty() && !specimenProcessResult.newSpecimenPids()
+        .isEmpty()) {
+      results.addAll(
+          digitalSpecimenService.createNewDigitalSpecimen(specimenProcessResult.newSpecimens(),
+              pidProcessResults));
     }
-    digitalMediaService.removeSpecimenRelationshipsFromMedia(digitalSpecimenRecords);
-    log.info("Persisting {} updated records to elastic", digitalSpecimenRecords.size());
-    try {
-      var bulkResponse = elasticRepository.indexDigitalSpecimen(
-          digitalSpecimenRecords.stream().map(UpdatedDigitalSpecimenRecord::digitalSpecimenRecord)
-              .toList());
-      if (!bulkResponse.errors()) {
-        handleSuccessfulElasticUpdate(digitalSpecimenRecords, mediaPidMap);
-      } else {
-        digitalSpecimenRecords = rollbackService.handlePartiallyFailedElasticUpdate(
-            digitalSpecimenRecords, mediaPidMap, bulkResponse);
-      }
-      var successfullyProcessedRecords = digitalSpecimenRecords.stream()
-          .map(UpdatedDigitalSpecimenRecord::digitalSpecimenRecord).collect(
-              Collectors.toSet());
-      log.info("Successfully updated {} digitalSpecimen records",
-          successfullyProcessedRecords.size());
-      annotationPublisherService.publishAnnotationUpdatedSpecimen(digitalSpecimenRecords);
-      gatherDigitalMediaObjectForUpdatedRecords(digitalSpecimenRecords);
-      return successfullyProcessedRecords;
-    } catch (IOException | ElasticsearchException e) {
-      log.error("Rolling back, failed to insert records in elastic", e);
-      rollbackService.rollbackUpdatedSpecimens(digitalSpecimenRecords, mediaPidMap, false, true);
-      return Set.of();
+    if (!specimenProcessResult.changedSpecimens().isEmpty()) {
+      results.addAll(digitalSpecimenService.updateExistingDigitalSpecimen(
+          specimenProcessResult.changedSpecimens(), pidProcessResults));
     }
+    return results;
   }
 
-  private void handleSuccessfulElasticUpdate(
-      Set<UpdatedDigitalSpecimenRecord> digitalSpecimenRecords,
-      Map<DigitalMediaKey, String> mediaPidMap) {
-    log.debug("Successfully indexed {} specimens", digitalSpecimenRecords);
-    var failedRecords = new HashSet<UpdatedDigitalSpecimenRecord>();
-    for (var digitalSpecimenRecord : digitalSpecimenRecords) {
-      var successfullyPublished = publishUpdateEvent(digitalSpecimenRecord);
-      if (!successfullyPublished) {
-        failedRecords.add(digitalSpecimenRecord);
-      }
+  private List<DigitalMediaRecord> processMediaResults(MediaProcessResult mediaProcessResult,
```

Comment:
IDE says return value is never used

Reply:
Yeah we will get to this when we get APIs. That was the mean reason the return was there. Not a priority now.
We will get other cases anyway (hit this with thinking about the virtual collections).
We will get cases where only the specimen is updated (through an annotation for example) and it doesn't contain the media. So we need a path to only update the specimen and probably the same for media. We might also need endpoints to push the specimen data with or without media. So we will need to hit this part again.

But for now I would not return anything so not to confuse us later.